README.md

```diff
@@ -526,17 +526,23 @@ For an example see [dinedal/go-sqlite3-extension-functions](https://github.com/d
 
 - Error: `database is locked`
 
-    When you get a database is locked. Please use the following options.
+    This means two different connections are trying to write to the database at
+    the same time.
 
-    Add to DSN: `cache=shared`
+    One way to fix this is to ensure your program limits the number of writers
+    to 1, for example by using `sync.Mutex` (it's fine to be have multiple
```

Comment:
Having multiple writers is actually not itself a problem. However, what can cause an issue even with write-ahead logging is if a connection tries to write after reading within a transaction.

```go
tx := database.Begin()
...
tx.QueryRow(`SELECT ID FROM Foo WHERE ...`)
...
tx.Exec(`UPDATE Foo ...`)
...
```

The problem here is that if the database gets modified between the `SELECT` and the `UPDATE` by another connection, there is no way for SQLite to properly retain the "isolation" property of ACID. This is because the interim modification may have changed the very data you read out. This issue can be avoided by always using immediate locking behavior (i.e., `BEGIN IMMEDIATE`) for any transactions that may ultimately write to the database. (It of course is also avoided by throttling the connection pool to one, provided that no other applications will also be writing to the same database.)
```